internal/pkg/template/templates/workloads/partials/cf/imported-alb-resources.yml

```diff
@@ -0,0 +1,142 @@
+  {{- $arn = $listener.ARN}}
+  {{- end}}
+  {{- end}}
+{{- end}}
+
+{{- if ne $arn ""}}
+HTTPRulePriorityAction:
+  Metadata:
+    'aws:copilot:description': 'A custom resource assigning priority for HTTP listener rules'
+  Type: Custom::RulePriorityFunction
+  Properties:
+    ServiceToken: !GetAtt RulePriorityFunction.Arn
+    RulePath: {{ fmtSlice (quoteSlice .ALBListener.RulePaths) }}
+    ListenerArn: {{$arn}}
+
+{{- $isHTTPS := .ALBListener.IsHTTPS}}
+{{- range $i, $rule := .ALBListener.Rules}}
+HTTPListenerRuleForImportedALB{{ if ne $i 0 }}{{ $i }}{{ end }}:
+  Metadata:
+    'aws:copilot:description': 'An HTTP listener rule for path `{{$rule.Path}}` that forwards HTTP traffic to your tasks'
+  Type: AWS::ElasticLoadBalancingV2::ListenerRule
+  Properties:
+    Actions:
+      - TargetGroupArn: !Ref TargetGroup{{ if ne $i 0 }}{{ $i }}{{ end }}
+        Type: forward
+    Conditions:
+      {{- if $rule.AllowedSourceIps}}
+      - Field: 'source-ip'
+        SourceIpConfig:
+          Values:
+          {{- range $sourceIP := $rule.AllowedSourceIps}}
+          - {{$sourceIP}}
+          {{- end}}
+      {{- end}}
+      {{- if $rule.Aliases}}
+      - Field: 'host-header'
+        HostHeaderConfig:
+          Values: {{fmtSlice (quoteSlice $rule.Aliases)}}
+      {{- else if $isHTTPS}}
+      - Field: 'host-header'
+        HostHeaderConfig:
+          Values:
+            - Fn::Join:
+                - '.'
+                - - !Ref WorkloadName
+                  - Fn::ImportValue:
+                      !Sub "${AppName}-${EnvName}-SubDomain"
+      {{- end}}
+      - Field: 'path-pattern'
+        PathPatternConfig:
+          Values:
+            {{- if eq $rule.Path "/" }}
+            - /*
+            {{- else }}
+            - {{ $rule.Path }}
+            - {{ $rule.Path }}/*
+            {{- end }}
+    ListenerArn: {{$arn}}
+    Priority: !GetAtt HTTPRulePriorityAction.Priority{{ if ne $i 0 }}{{ $i }}{{ end }}
+  {{- end}}
+{{- end }}
+
+# If two listeners, redirect port 80 to 443.
+{{- if $.ImportedALB.HasTwoListeners }}
+{{- $port80arn := ""}}
+{{- range $listener := $.ImportedALB.Listeners}}
+  {{- if eq $listener.Port 80}}
+  {{- $port80arn = $listener.ARN}}
+  {{- end}}
+{{- end}}
+{{- $isHTTPS := .ALBListener.IsHTTPS}}
+{{- if ne $port80arn ""}}
+HTTPRedirectRulePriorityAction:
+  Metadata:
+    'aws:copilot:description': 'A custom resource assigning priority for HTTP listener rules'
+  Type: Custom::RulePriorityFunction
+  Properties:
+    ServiceToken: !GetAtt RulePriorityFunction.Arn
+    RulePath: {{ fmtSlice (quoteSlice .ALBListener.RulePaths) }}
+    ListenerArn: {{$port80arn}}
+
+{{- range $i, $rule := .ALBListener.Rules}}
+HTTPListenerRedirectRuleForImportedALB{{ if ne $i 0 }}{{ $i }}{{ end }}:
+  Metadata:
+    'aws:copilot:description': 'An HTTP listener rule that redirects port 80 to port 443'
+  Type: AWS::ElasticLoadBalancingV2::ListenerRule
+  Properties:
+    Actions:
+      - Type: redirect
```

Comment:
maybe not in this release, but we might want to consider the `http.redirect_to_https` manifest field like in https://github.com/aws/copilot-cli/blob/baf2a7849407b975ace780479248cde9b441d3ff/internal/pkg/template/templates/workloads/partials/cf/https-listener.yml#L85